src/cpp/include/openvino/genai/rag/text_embedding_pipeline.hpp

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <filesystem>
+#include <optional>
+#include <variant>
+
+#include "openvino/genai/tokenizer.hpp"
+
+namespace ov {
+namespace genai {
+
+using EmbeddingResult = std::variant<std::vector<float>, std::vector<int8_t>, std::vector<uint8_t>>;
+
+class OPENVINO_GENAI_EXPORTS TextEmbeddingPipeline {
+public:
+    enum class OPENVINO_GENAI_EXPORTS PoolingType {
+        /**
+         * @brief First token embeddings
+         */
+        CLS = 0,
+        /**
+         * @brief The average of all token embeddings
+         */
+        MEAN = 1,
+    };
+
+    struct OPENVINO_GENAI_EXPORTS Config {
+        std::optional<size_t> max_length;
+        PoolingType pooling_type = PoolingType::CLS;
+        bool normalize = false;
+        std::optional<std::string> query_instruction;
+        std::optional<std::string> embed_instruction;
+
+        Config() = default;
+        explicit Config(const ov::AnyMap& properties);
+    };
+
+    /**
+     * @brief Constructs a pipeline from xml/bin files, tokenizer and configuration in the same dir.
+     *
+     * @param models_path Path to the dir model xml/bin files, tokenizer
+     * @param device optional device
+     * @param plugin_config optional plugin_config properties
+     */
+    TextEmbeddingPipeline(const std::filesystem::path& models_path,
+                          const std::string& device,
+                          const std::optional<Config>& config = std::nullopt,
+                          const ov::AnyMap& properties = {});
+
+    TextEmbeddingPipeline(const std::filesystem::path& models_path,
+                          const std::string& device,
+                          const ov::AnyMap& properties);
+
+    template <typename... Properties,
+              typename std::enable_if<ov::util::StringAny<Properties...>::value, bool>::type = true>
+    TextEmbeddingPipeline(const std::filesystem::path& models_path,
+                          const std::string& device,
+                          Properties&&... properties)
+        : TextEmbeddingPipeline(models_path, device, ov::AnyMap{std::forward<Properties>(properties)...}) {}
+
+    /**
+     * @brief Computes embeddings for a vector of texts
```

Comment:
please, add documentation for parameters 

and other methods / API entities within this file.

Reply:
Docstrings added